Building DAG of jobs...
Full Traceback (most recent call last):
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/dag.py", line 2070, in collect_potential_dependencies
    yield PotentialDependency(file, known_producers[file], True)
                                    ~~~~~~~~~~~~~~~^^^^^^
KeyError: 'results/align/bam_original/WT_REP1_control.Aligned.out.bam'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/jobs.py", line 183, in new
    obj = self.cache[key]
          ~~~~~~~~~~^^^^^
KeyError: ('align', ('sample', 'WT_REP1_control'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/cli.py", line 2103, in args_to_api
    dag_api.execute_workflow(
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/api.py", line 594, in execute_workflow
    workflow.execute(
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/workflow.py", line 1081, in execute
    self._build_dag()
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/workflow.py", line 1037, in _build_dag
    async_run(self.dag.init())
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/common/__init__.py", line 94, in async_run
    return asyncio.run(coroutine)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/dag.py", line 179, in init
    job = await self.update([job], progress=progress, create_inventory=True)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/dag.py", line 1013, in update
    raise exceptions[0]
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/dag.py", line 970, in update
    await self.update_(
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/dag.py", line 1055, in update_
    potential_dependencies = [
                             ^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/dag.py", line 2088, in collect_potential_dependencies
    await file2jobs(
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/dag.py", line 2181, in file2jobs
    await self.new_job(
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/dag.py", line 1927, in new_job
    job = await self.job_factory.new(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/jobs.py", line 185, in new
    obj = await new()
          ^^^^^^^^^^^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/jobs.py", line 158, in new
    obj = new_job()
          ^^^^^^^^^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/jobs.py", line 151, in <lambda>
    new_job = lambda: Job(
                      ^^^^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/jobs.py", line 254, in __init__
    ) = self.rule.expand_input(self.wildcards_dict, groupid=groupid)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/rules.py", line 795, in expand_input
    incomplete = self._apply_wildcards(
                 ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucianhu/micromamba/envs/rna_snakemake/lib/python3.12/site-packages/snakemake/rules.py", line 731, in _apply_wildcards
    raise WorkflowError(
snakemake_interface_common.exceptions.WorkflowError: Cannot combine named input file (name fq_trim) with unpack() (rule align, line 241, /home/lucianhu/rna_snakemake/test_ting/Snakefile)

WorkflowError in rule align in file /home/lucianhu/rna_snakemake/test_ting/Snakefile, line 141:
Cannot combine named input file (name fq_trim) with unpack() (rule align, line 241, /home/lucianhu/rna_snakemake/test_ting/Snakefile)
